.pipelines/e2e.yml

```diff
@@ -10,47 +10,79 @@ resources:
 # Azure DevOps Pipeline running e2e tests
 variables:
 - template: vars.yml
-jobs:
-- job: E2E
-  timeoutInMinutes: 180
-  pool:
-    name: ARO-CI
-    demands: go-1.16
-  steps:
-  - template: ./templates/template-checkout.yml
-  - template: ./templates/template-az-cli-login.yml
-    parameters:
-      azureDevOpsJSONSPN: $(aro-v4-e2e-devops-spn)
-  - script: |
-      az account set -s $AZURE_SUBSCRIPTION_ID
-      export SECRET_SA_ACCOUNT_NAME=$(SECRET_SA_ACCOUNT_NAME)
-      make secrets
-      . secrets/env
-      echo "##vso[task.setvariable variable=RP_MODE]$RP_MODE"
-    displayName: 🔑 Downloading certificates and secrets from storage account
-    name: setEnv
-  - template: ./templates/template-push-images-to-acr.yml
-    parameters:
-      rpImageACR: $(RP_IMAGE_ACR)
-      buildCommand: publish-image-aro
-  - script: |
-      set -e
-      set -o pipefail
-
-      . secrets/env
-
-      az account set -s $AZURE_SUBSCRIPTION_ID
-
-      set -x
-      . ./hack/e2e/run-rp-and-e2e.sh
-      trap 'set +e; kill_rp; clean_e2e_db' EXIT
-
-      deploy_e2e_db
-
-      run_rp
-      validate_rp_running
-      register_sub
-
-      export CI=true
-      make test-e2e
-  - template: ./templates/template-az-cli-logout.yml
+
+# Run the test suite and collect must-gather
+stages:
+- stage: E2ESuite
+  jobs:
+  - job: Execute_Tests
+    timeoutInMinutes: 180
+    pool:
+      name: ARO-CI
+      demands: go-1.16
+    steps:
+    - template: ./templates/template-checkout.yml
+    - template: ./templates/template-az-cli-login.yml
+      parameters:
+        azureDevOpsJSONSPN: $(aro-v4-e2e-devops-spn)
+    - template: ./templates/template-certificates-and-secrets.yml
+    - template: ./templates/template-push-images-to-acr.yml
+      parameters:
+        rpImageACR: $(RP_IMAGE_ACR)
+        buildCommand: publish-image-aro
+    - script: |
+        set -o pipefail
+
+        . secrets/env
+        az account set -s $AZURE_SUBSCRIPTION_ID
+
+        set -x
+        . ./hack/e2e/run-rp-and-e2e.sh
+        trap 'set +e; kill_rp; clean_e2e_db' EXIT
+
+        deploy_e2e_db
+        run_rp
+        validate_rp_running
+        register_sub
+
+        export CI=true
+        export LOCATION=eastus
+        export BUILD_BUILDID=$(Build.BuildId)
+        export CLUSTER="v4-e2e-V$BUILD_BUILDID-$LOCATION"
+        export DATABASE_NAME="v4-e2e-V$BUILD_BUILDID-$LOCATION"
+        . secrets/env
+        . ./hack/e2e/run-rp-and-e2e.sh
```

Comment:
@karanmagdani attempted to re-source these environment variables beforehand to ensure that we weren't missing any.

Reply:
The `secrets/env` file contains an `export DATABASE_NAME="$USER"`, so I'd say you need to `export DATABASE_NAME="v4-e2e-V$BUILD_BUILDID-$LOCATION"` after sourcing the secrets.

There is also a line in the secrets that should be removed according to the comment by Jim, but would need some investigation first:
```sh
# JM(18/11/2020): remove this after https://github.com/Azure/ARO-RP/pull/1174
COSMOSDB_ACCOUNT="$RESOURCEGROUP"
```

Reply:
@darthhexx it was an issue with secrets after all - needed to `make secrets`, and now it works :)